rls/src/build/cargo.rs

```diff
@@ -206,8 +183,8 @@ fn run_cargo_ws(
 
     // Since the Cargo build routine will try to regenerate the unit dep graph,
     // we need to clear the existing dep graph.
-    compilation_cx.lock().unwrap().build_plan =
-        BuildPlan::Cargo(CargoPlan::with_packages(manifest_path, pkg_names));
```

Comment:
I'm not 100% sure `manifest_path` is always equivalent to `ws.root()`? While we construct a `Workspace` from a `manifest_path`, the `Workspace::root` returns [path to a root manifest](https://github.com/rust-lang/cargo/blob/1e6828485eea0f550ed7be46ef96107b46aeb162/src/cargo/core/workspace.rs#L241-L244) if we're initialized in a workspace member directory.
As such, I'd bring back `manifest_path` and don't change semantics together with a cleanup code